FaunaDB.Client/Reflection/Encoder.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using FaunaDB.Query;
+using FaunaDB.Types;
+using FaunaDB.Attributes;
+
+namespace FaunaDB.Reflection
+{
+    public static class Encoder
+    {
+        public static Expr EncodeObject(object obj) =>
```

Comment:
Actually it's possible given I had to "encode" fields/properties of any type and that function is called recursively for each field/property if you pass in a collection of any type it'll encode as well